changes/2020-07-06_clarify-streaming-encrypt-decrypt/change.md

```diff
@@ -0,0 +1,154 @@
+[//]: # "Copyright Amazon.com Inc. or its affiliates. All Rights Reserved."
+[//]: # "SPDX-License-Identifier: CC-BY-SA-4.0"
+
+# Clarify Streaming Encrypt and Decrypt
+
+## Affected Features
+
+This change affects no features.
+
+## Affected Specifications
+
+This serves as a reference of all specification documents that this change affects.
+
+| Specification                           |
+| --------------------------------------- |
+| [Encrypt](../../client-apis/encrypt.md) |
+| [Decrypt](../../client-apis/decrypt.md) |
+
+## Affected Implementations
+
+This change affects no implementations.
+
+## Definitions
+
+### Conventions used in this document
+
+The key words "MUST", "MUST NOT", "REQUIRED", "SHALL", "SHALL NOT", "SHOULD", "SHOULD NOT", "RECOMMENDED", "MAY", and "OPTIONAL"
+in this document are to be interpreted as described in [RFC 2119](https://tools.ietf.org/html/rfc2119).
+
+## Summary
+
+We specify how to perform encryption and decryption in
+the streaming context,
+specifically specifying what processing MUST occur when,
+and what conditions are REQUIRED before releasing output.
+
+## Out of Scope
+
+Any feature updates are out of scope for this change.
+
+## Motivation
+
+The Encrypt and Decrypt specifications do not currently make it clear how these operations
+MUST be performed in order to support streaming use cases.
+This change clarifies how to operate on partial data
+and what conditions are REQUIRED to release output.
+
+## Drawbacks
+
+There SHOULD be no drawbacks for specifying the streaming use case.
+
+## Security Implications
+
+There SHOULD be no security implications for specifying the streaming use case.
+
+## Operational Implications
+
+There SHOULD be no operational implications for specifying the streaming use case.
+
+## Guide-level Explanation
+
+The following sections describe how this change is updating the specification
+to answer important questions.
+
+### How should the operation process data in parts?
+
+This change rewrites these specifications to
+better specify how these operations are allowed to process data in parts.
+Specifically, we attempt to be completely language agnostic and describe streaming as a way for
+bytes to move sequentially, over time.
+
+With this framing, the things the encrypt and decrypt operations care about are:
+
+- Is the caller allowed to send any more data?
+- What data is currently available to me?
+
+In the context of the encrypt or decrypt operation, we specify the conditions
+for when the operation MAY or MUST perform some processing.
+These conditions include how many input bytes are available,
+whether more bytes MAY be available,
+at what step in the operation you are in,
+and the result of various calculations.
+
+Additionally, we use similar conditions to specify when the output of the operation MAY be released.
+
+This change SHOULD NOT introduce any language which is implementation specific for
+a specific streaming interface.
```

Comment:
Couldn't make sense of this sentence. Do you mean "SHOULD NOT introduce any language-specific implementation..."?